sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -791,53 +805,69 @@ private BasicDocumentRevision deleteDocumentInQueue(SQLDatabase db, final String
 
         CouchUtils.validateRevisionId(prevRevId);
 
-                    BasicDocumentRevision deletedRevision = null;
+        BasicDocumentRevision deletedRevision = null;
+            BasicDocumentRevision preRevision;
+        try {
+            preRevision = getDocumentInQueue(db, docId, prevRevId);
+        }  catch (AttachmentException e){
+            throw new DocumentNotFoundException(e);
+        }
```

Comment:
I've seen this flipping of exception type a few times now. 

Are there any cases where, when `getDocumentInQueue` throws an `AttachmentException` that it's not converted into `DocumentNotFoundException` exception by the caller? If not, do the conversion inside `getDocumentInQueue`.